corehq/apps/dump_reload/sql/load.py

```diff
@@ -157,8 +168,12 @@ def load_data_for_db(db_alias):
                     continue
                 model_counter.update([Model])
                 try:
-                    obj.save(using=db_alias)
+                    # Force insert here to prevent Django from attempting to do an update.
+                    # We want to ensure that if there is already data in the DB that we don't
+                    # save over it and rather error out.
+                    obj.save(using=db_alias, force_insert=True)
```

Comment:
Note this change. Without this Django will just update any existing models if they exist in the DB (with the same primary key).

Reply:
Does this raise an exception if `obj` already exists, or does it just not save?

So if we want to "tread carefully" when resuming loading, I imagine we'd want to update existing records up to a point. Do you think it would it make sense to use `force_update` until the record is not found, and then switch to `force_insert`?

Reply:
This will error if the record already exists. It could be possible to use `force_update` but since we don't have any mechanism for resume at present I'm leaving that as future work. I have another PR I'm working on which provides a pluggable strategy for saving which could maybe be used.

Reply:
I'm punting on the "resumable load" work for now, though I do think there are some possibilities.  @snopoke let me know if you want to talk about that at all.